src/api/cvc4cpp.h

```diff
@@ -0,0 +1,2357 @@
+     */
+    bool isPredicate() const;
+
+    /**
+     * Is this a tuple sort?
+     * @return true if the sort is a tuple sort
+     */
+    bool isTuple() const;
+
+    /**
+     * Is this a record sort?
+     * @return true if the sort is a record sort
+     */
+    bool isRecord() const;
+
+    /**
+     * Is this an array sort?
+     * @return true if the sort is a array sort
+     */
+    bool isArray() const;
+
+    /**
+     * Is this a Set sort?
+     * @return true if the sort is a Set sort
+     */
+    bool isSet() const;
+
+    /**
+     * Is this a sort kind?
+     * @return true if this is a sort kind
+     */
+    bool isSort() const;
+
+    /**
+     * Is this a sort constructor kind?
+     * @return true if this is a sort constructor kind
+     */
+    bool isSortConstructor() const;
+
+    /**
+     * @return the underlying datatype of a datatype sort
+     */
+    Datatype getDatatype() const;
+
+    /**
+     * Instantiate a parameterized datatype/sort sort.
+     * Create sorts parameter with Solver::mkParamSort().
+     * @param params the list of sort parameters to instantiate with
+     */
+    Sort instantiate(const std::vector<Sort>& params) const;
+
+    /**
+     * Output a string representation of this sort to a given stream.
+     * @param out the output stream
+     */
+    void toStream(std::ostream& out) const;
+
+    /**
+     * @return a string representation of this sort
+     */
+    std::string toString() const;
+
+  private:
+    /* The interal type wrapped by this sort. */
+    CVC4::Type* d_type;
+};
+
+/**
+ * Serialize a sort to given stream.
+ * @param out the output stream
+ * @param s the sort to be serialized to the given output stream
+ * @return the output stream
+ */
+std::ostream& operator<< (std::ostream& out, const Sort& s) CVC4_PUBLIC;
+
+/**
+ * Hash function for Sorts.
+ */
+struct CVC4_PUBLIC SortHashFunction
+{
+  size_t operator()(const Sort& s) const;
+};
+
+
+/* -------------------------------------------------------------------------- */
+/* Term                                                                       */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * A CVC4 Term.
+ */
+class CVC4_PUBLIC Term
+{
+  friend class Solver;
+  friend struct TermHashFunction;
+
+  public:
+    /**
+     * Constructor.
+     */
+    Term();
+
+    /**
+     * Constructor.
+     * @param e the internal expression that is to be wrapped by this term
+     * @return the Term
+     */
+    Term(const CVC4::Expr& e);
+
+    /**
+     * Copy constructor.
+     */
+    Term(const Term& t);
+
+    /**
+     * Destructor.
+     */
+    ~Term();
+
+    /**
+     * Assignment operator, makes a copy of the given term.
+     * Both terms must belong to the same solver object.
+     * @param t the term to assign
+     * @return the reference to this term after assignment
+     */
+    Term& operator=(const Term& t);
+
```

Comment:
Consider adding an operator=(Term&&).

Reply:
Only for Term? Or also for the other wrapper classes?

Reply:
Term is the most interesting as this will be the heaviest used.

Reply:
See above.
Done.